[ABLD-169] Keep our own copy of cacert.pem

aiuto opened the pull request:
- Maintain our own copy of certs rather than downloading during the build
- Include text in the BUILD file about how we check for new upstream versions.
- Add explanation of why we have this.
- Add tooling to install via //packags/install_dir/embedded:everything.
- Not used yet to minimize rollback surface.  The followup PR will put it in place.

### Followup PR
- Add to  //packags/install_dir/embedded:everything.
- Delete cacerts.rb

https://datadoghq.atlassian.net/browse/ABLD-169

alopezz commented on omnibus/config/software/cacerts.rb:
```diff
@@ -25,33 +25,21 @@
-    copy "#{project_dir}/cacert.pem", "#{python_3_embedded}/ssl/certs/cacert.pem"
-    copy "#{project_dir}/cacert.pem", "#{python_3_embedded}/ssl/cert.pem"
+    mkdir "#{python_3_embedded}/embedded/ssl/cacerts"
+    copy "#{Omnibus::Config.project_root()}/../deps/cacerts/cacert.pem", "#{python_3_embedded}/embedded/ssl"
```
We should probably be using the `source` directive instead, like we do for the [datadog-agent definition](https://github.com/DataDog/datadog-agent/blob/303fe6eb196422978696d628787840834743004a/omnibus/config/software/datadog-agent.rb#L31).

```ruby
source path: '../deps/cacerts'
```

And then this bit would become 
```suggestion
    copy "cacert.pem", "#{python_3_embedded}/embedded/ssl"
```

And the `license_file` above could also point to the local file that you vendored.

aiuto replied:
Interesting.  I will try that. But I wonder about source having an option to exclude files. That seems to indicate a copy or symlink tree, rather than direct reference unpacked git repo.  I'll do some research.

aiuto replied:
I don't want the extra copy, so I left it mostly as is. Just using the same cwd construct that you us in the run_as_root command.

Of course, CI may tell me I have to back it out.

aiuto commented:
Update on this.
I consulted with security folks and they agree with the concept.  They are working on finding the right role to review future changes to cacert.pem within their team.

aiuto committed:
[ABLD-169] Keep our own copy of cacert.pem

- Replace omnibus fetch from upstream with that static copy.
- Include text in the BUILD file about how we check for new upstream versions.
- Add explanation of why we have this.
- use cwd to make it a little cleaner
- update cert to 2025-09-09

 It doesn't matter if it is ugly or not. We are going to delete it this quarter anyway.

aiuto commented on deps/cacerts/BUILD.bazel:
```diff
@@ -0,0 +1,125 @@
+
+This list is used by our python integrations, and possibly other bits.
+It forms the root of the web of trust. From a reliability, and security
+point of view, it is best if our code uses a set that we have veted
```
done

aiuto committed:
update file

aiuto committed:
fix the test to pick up the filst the rightgway

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
and use the certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
do not duplicate install in openssl

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
fix the paths

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
update to 2026-03-19

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
Merge branch 'main' into aiuto/certs

aiuto committed:
freaking test ...

aiuto committed:
Merge branch 'main' into aiuto/certs
